source/loader/layers/validation/ur_leak_check.hpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include "ur_validation_layer.hpp"
+
+#include <execinfo.h>
+#include <mutex>
+#include <shared_mutex>
+#include <unordered_map>
+#include <utility>
+
+#define MAX_BACKTRACE_FRAMES 64
+
+namespace validation_layer {
+class RefCountContext {
+  private:
+    struct RefRuntimeInfo {
+        int64_t refCount;
+        char **originBacktrace;
```

Comment:
You might want to wrap originBacktrace in std::unique_ptr here, you wouldn't need to call `clearRefInfo` manually and it would also prevent leaks in case of unhandled exceptions.

Or alternatively, you could just convert `backtraceStrings` you get from `backtrace_symbols` to std::string inside `createRefInfo` and just store a string here.

Reply:
I've changed `char **` into a `std::vector` of `std::string`.